### PR TITLE
[New feature - Code & Docs] Scenarios management

### PR DESCRIPTION
**Pull request by assouktim**
**Scenario Management** allows creation of scenarios representing complex interactions between user and bot. It allows the user's initial intention to be used to determine an objective to be achieved and to question the user in order to collect information that will allow the user's request to be answered in an optimal manner. 

The core of feature is managed by "**Dialog Manager**" : advanced dialogue manager allowing, via the use of the state machine combined with constraint-based graph resolution, to optimise the user path in the scenario.

For more information : [Docs](https://github.com/CreditMutuelArkea/tock/blob/contribution/scenarios/v1.0/docs/_fr/user/studio/scenarios-management.md)

See also : 
tock-docker PR : [https://github.com/theopenconversationkit/tock-docker/pull/29](https://github.com/theopenconversationkit/tock-docker/pull/29)

### CODE REVIEW

File: bot/admin/server/src/main/kotlin/BotAdminVerticle.kt [hunk 1]
```diff
@@ -700,17 +706,55 @@ open class BotAdminVerticle : AdminVerticle() {
...
+            "/i18n/search",
+            setOf(botUser, faqBotUser),
+            simpleLogger("search i18n with ids")
+        ) { _: RoutingContext, query: I18nLabelSummarySearchQuery ->
+            if (query.i18nIds.isEmpty()) {
+                badRequest("no filter found on i18nIds")
+            } else {
+                return@blockingJsonPost BotI18nLabels(
+                    i18n.getLabelsByIds(query.i18nIds)
+                        .ifEmpty { notFound() }
```
Comment (djowalker) [inline bot/admin/server/src/main/kotlin/BotAdminVerticle.kt:733]: May cause trouble when importing scenario, will be fixed.

File: bot/admin/server/src/main/kotlin/mapper/ScenarioExceptionManager.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    fun <O> catch(fallibleSection: () -> O): O {
+        try {
+            return fallibleSection.invoke()
+        } catch(ex : ScenarioException) {
+            logger.error(ex.message)
+            throw ex.toRestException()
+        }
+    }
+
+    private fun ScenarioException.toRestException(): RestException {
```
Comment (vsct-jburet) [inline bot/admin/server/src/main/kotlin/mapper/ScenarioExceptionManager.kt:49]: What do you think of the idea to add this function directly in the ScenarioException class ?

File: bot/engine/src/main/kotlin/engine/TockConnectorController.kt [hunk 3]
```diff
@@ -80,6 +90,11 @@ internal class TockConnectorController constructor(
                 .apply {
                     logger.info { "Register connector $connector for bot $bot" }
                     connector.register(this)
+
+                    if(debugEnabled) {
+                        // FIXME (WITH DERCBOT-321)
+                        exposePythonLog()
```
Comment (vsct-jburet) [inline bot/engine/src/main/kotlin/engine/TockConnectorController.kt:96]: Do you want to keep this in the PR ?

**Comment by assouktim** [comment]
This topic has been overtaken by the arrival of generative AI on Tock.

**Pull request closed without merging**